src/util/BaseEntityUtil.java

```diff
@@ -1,18 +1,33 @@
 package util;
 
 import entities.BaseEntity;
+import entities.Cashier;
+import entities.Client;
+import java.util.Map;
 import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
 
 public final class BaseEntityUtil {
+
+  public static final Map<Class<?>, Function<?, ?>> ENTITY_KEY_TO_GENERATOR_FUNCTION_MAP =
+      Map.of(Long.class, (Long currentId) -> currentId + 1);
+  public static final Map<Class<?>, BiFunction<?, ?, ?>> ENTITY_TO_FETCH_FUNCTION_MAP =
+      Map.of(
+          Client.class,
+          (target, source) -> ClientUtil.fetchClient((Client) target, (Client) source),
```

Comment:
maybe ` (Client target, Client source) -> ClientUtil.fetchClient(target, source)` is cleaner?

Reply:
solved